build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,239 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Context, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import clonedeep from "lodash.clonedeep";
+import { ResetMode } from "simple-git";
+
+import { packageOrReleaseGroupArg as baseArg, findPackageOrReleaseGroup } from "../../args";
+import { BaseCommand } from "../../base";
+import { Repository, getPreReleaseDependencies } from "../../lib";
+import { isReleaseGroup, ReleaseGroup, ReleasePackage } from "../../releaseGroups";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+const checkNoLocalChanges: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
+		};
+	}
+
+	return { result: true };
+};
+
+const checkDepsInstalled: CheckFunction = async (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+): Promise<CheckResult> => {
+	const packagesToCheck = isReleaseGroup(rgOrPkg)
+		? context.packagesInReleaseGroup(rgOrPkg)
+		: // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		  [context.fullPackageMap.get(rgOrPkg)!];
+
+	async function checkDeps(pkg: Package): Promise<boolean> {
+		const depsAreInstalled = await pkg.checkInstall(print);
+		// console.warn(`deps for ${pkg.name}: ${depsAreInstalled}`);
+		return depsAreInstalled;
+		// return pkg.checkInstall(false);
+	}
+
+	// eslint-disable-next-line unicorn/no-array-method-this-argument, @typescript-eslint/no-misused-promises
```

Comment:
Ideally we would write code in a way that doesn't violate our lint rules, or in cases where it is actually ok, include a comment with the lint suppression saying why this case is ok. Offhand as someone reading the code I don't know why these lint rules exist (I'd need to look that up) so its hard for me to know if this code has the issues those rules were designed to prevent without a comment to explain.